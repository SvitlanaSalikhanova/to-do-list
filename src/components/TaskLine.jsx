import React from 'react';
import Check from './Check';
import DeleteButton from './DeleteButton';
import TaskLineCss from '../styles/TaskLine.module.scss';

export default function TaskLine({ item, style }) {
    const [showDeleteButton, setShowDeleteButton] = React.useState(false);
    return (
        <div
            className={TaskLineCss.taskLine}
            onMouseEnter={() => setShowDeleteButton(true)}
            onMouseLeave={() => setShowDeleteButton(false)}
            style={style}
        >
            <Check item={item} />

            <div
                className={
                    item.isDone
                        ? TaskLineCss.completedTaskText
                        : TaskLineCss.uncompletedTaskText
                }
            >
                {item.text}
            </div>

            {showDeleteButton && <DeleteButton item={item} />}
        </div>
    );
}
