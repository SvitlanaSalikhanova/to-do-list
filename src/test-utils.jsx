import React from 'react';
import { render } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import tasksReducer from './redux/TasksSlice';
import filterReducer from './redux/FilterSlice';

function renderWithProvider(
    ui,
    {
        preloadedState = {},
        store = configureStore({
            reducer: {
                tasks: tasksReducer,
                filter: filterReducer,
            },
            preloadedState,
        }),
        ...renderOptions
    } = {},
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>;
    }

    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

function providerWrapper({ children }, preloadedState = {}) {
    const store = configureStore({
        reducer: {
            tasks: tasksReducer,
            filter: filterReducer,
        },
        preloadedState,
    });
    return <Provider store={store}>{children}</Provider>;
}

export { providerWrapper, renderWithProvider };
