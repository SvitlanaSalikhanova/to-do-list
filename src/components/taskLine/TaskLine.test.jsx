import React from 'react';
import renderer from 'react-test-renderer';
import { screen } from '@testing-library/react';
import TaskLine from './TaskLine';
import { renderWithProvider, providerWrapper } from '../../test-utils';

const UncompletedItem = {
    id: 1,
    text: 'text',
    isDone: false,
};

const CompletedItem = {
    id: 1,
    text: 'text',
    isDone: true,
};

describe('TaskLine component tests', () => {
    it('TaskLine for Completed task Matches Snapshot', () => {
        const domTree = renderer.create(
            providerWrapper(<TaskLine item={UncompletedItem} />),
        ).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    it('TaskLine for Uncompleted task Matches Snapshot', () => {
        const domTree = renderer.create(
            providerWrapper(<TaskLine item={CompletedItem} />),
        ).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    it('TaskLine checkbox for completed task is checked', () => {
        renderWithProvider(<TaskLine item={CompletedItem} />);
        const check = screen.getByRole('checkbox');
        expect(check).toBeInTheDocument();
        expect(check.checked).toBe(true);
    });

    it('TaskLine checkbox for uncompleted task is unchecked', () => {
        renderWithProvider(<TaskLine item={UncompletedItem} />);
        const check = screen.getByRole('checkbox');
        expect(check).toBeInTheDocument();
        expect(check.checked).toBe(false);
    });

    it('TaskLine description renders', () => {
        renderWithProvider(<TaskLine item={UncompletedItem} />);
        const description = screen.getByText('text');
        expect(description).toBeInTheDocument();
    });

    it('TaskLine deleteButton renders', () => {
        renderWithProvider(<TaskLine item={CompletedItem} />);
        const button = screen.getByRole('button');
        expect(button).toBeInTheDocument();
    });
});
