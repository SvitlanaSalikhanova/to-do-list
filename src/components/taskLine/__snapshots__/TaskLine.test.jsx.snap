// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskLine component tests TaskLine for Completed task Matches Snapshot 1`] = `null`;

exports[`TaskLine component tests TaskLine for Uncompleted task Matches Snapshot 1`] = `null`;
